test/e2e/prometheus_test.go

```diff
@@ -254,8 +256,8 @@ func testMetricsFromPrometheusServer(t *testing.T, data *TestData, prometheusJob
 	hostIP, nodePort := getPrometheusEndpoint(t, data)
 
 	// Build the Prometheus query URL
-	path := url.PathEscape("match[]={job=\"" + prometheusJob + "\"}")
-	queryUrl := fmt.Sprintf("http://%s:%d/api/v1/series?%s", hostIP, nodePort, path)
+	path := url.PathEscape("match_target={job=\"" + prometheusJob + "\"}")
+	queryUrl := fmt.Sprintf("http://%s:%d/api/v1/targets/metadata?%s", hostIP, nodePort, path)
```

Comment:
please add a comment to explain that this API (`GET /api/v1/targets/metadata`) has been available since Prometheus v2.4.0, but is still considered experimental and may break in the future.

Reply:
Added comment.